did_peer/src/key/verification_method.rs

```diff
@@ -0,0 +1,228 @@
+use did_doc::schema::verification_method::{
+    IncompleteVerificationMethodBuilder, VerificationMethod, VerificationMethodType,
+};
+use did_parser::{Did, DidUrl};
+
+use crate::{error::DidPeerError, peer_did_resolver::options::PublicKeyEncoding};
+
+use super::{Key, SupportedKeyType};
+
+pub fn get_verification_methods_by_key(
+    key: &Key,
+    did: &Did,
+    public_key_encoding: &PublicKeyEncoding,
+) -> Result<Vec<VerificationMethod>, DidPeerError> {
+    let id = DidUrl::from_fragment(key.prefixless_fingerprint().chars().take(8).collect::<String>())?;
+    let vm_type = match key.key_type() {
+        SupportedKeyType::Ed25519 => VerificationMethodType::Ed25519VerificationKey2020,
+        SupportedKeyType::Bls12381g1 => VerificationMethodType::Bls12381G1Key2020,
+        SupportedKeyType::Bls12381g2 => VerificationMethodType::Bls12381G2Key2020,
+        SupportedKeyType::X25519 => VerificationMethodType::X25519KeyAgreementKey2020,
+        SupportedKeyType::P256 => VerificationMethodType::JsonWebKey2020,
+        SupportedKeyType::P384 => VerificationMethodType::JsonWebKey2020,
+        SupportedKeyType::P521 => VerificationMethodType::JsonWebKey2020,
+        SupportedKeyType::Bls12381g1g2 => {
+            let g1_key = Key::new(key.key()[..48].to_vec(), SupportedKeyType::Bls12381g1)?;
+            let g2_key = Key::new(key.key()[48..].to_vec(), SupportedKeyType::Bls12381g2)?;
+            return Ok(build_verification_methods_from_bls_multikey(
+                g1_key,
+                g2_key,
+                id,
+                did.to_owned(),
+                public_key_encoding,
+            ));
+        }
+    };
+    Ok(build_verification_methods_from_type_and_key(
+        vm_type,
+        key,
+        id,
+        did.to_owned(),
+        public_key_encoding,
+    ))
+}
+
+pub fn get_key_by_verification_method(vm: &VerificationMethod) -> Result<Key, DidPeerError> {
+    let key_type = match vm.verification_method_type() {
+        VerificationMethodType::Ed25519VerificationKey2018 | VerificationMethodType::Ed25519VerificationKey2020 => {
+            SupportedKeyType::Ed25519
+        }
+        VerificationMethodType::Bls12381G1Key2020 => SupportedKeyType::Bls12381g1,
+        VerificationMethodType::Bls12381G2Key2020 => SupportedKeyType::Bls12381g2,
+        VerificationMethodType::X25519KeyAgreementKey2019 | VerificationMethodType::X25519KeyAgreementKey2020 => {
+            SupportedKeyType::X25519
+        }
+        t @ _ => return Err(DidPeerError::UnsupportedVerificationMethodType(t.to_owned())),
+    };
+    Key::new(vm.public_key().key_decoded()?, key_type)
+}
```

Comment:
What do you think of making this a method on the `VerificationMethod` type?

Reply:
The method returns `Key`, which is defined currently in `did_peer`, and both are only used in `did_peer`. I am not convinced that `did_peer` is the ideal place for either of them, but put them here for now since they are not used anywhere else.

Reply:
Hm, fair.